### PR TITLE
Added RSA key file support

### PR DESCRIPTION
**Pull request by ggruening**
Hello everybody,

some applications (like `mysql` for example) need key files in RSA format. I prepared RSA key conversion using openssl. You can enable it by setting `CONVERT_KEY2RSA=yes`. 
If you think this is useful I'd update the README, too.

Best regards, and thanks for this fantastic tool.

Gregor

### CODE REVIEW

**Comment by kereis** [comment]
Hey, thank you for your contribution.

I have 2 suggestions:
1. Renaming `CONVERT_KEY2RSA` to `CONVERT_KEY_TO_RSA`.
2. As the new flag will affect all keys involved in the extraction process, we should consider renaming it into `CONVERT_KEYS_TO_RSA` (with `KEYS` having a plural 's')

**Comment by ggruening** [comment]
I gladly took up your suggestions, see my second commit.

File: bin/dump.sh [hunk 1]
```diff
@@ -157,6 +160,28 @@ combine_pkcs12() {
...
+convert_keys_to_rsa() {
+  if [[ "${CONVERT_KEYS_TO_RSA}" != yes ]]; then
+    return
+  fi
+
+  if [[ -z "${DOMAIN}" || "${#DOMAINS[@]}" -gt 1 ]]; then
+    local outputdir_subdirs=(${outputdir}/*/)
+    for subdir in "${outputdir_subdirs[@]}"; do
+      local i=$(basename "${subdir}" /)
+      if [[ -f ${outputdir}/${i}/cert.pem && -f ${outputdir}/${i}/key.pem ]]; then
```
Comment (kereis) [inline bin/dump.sh:172]:
```suggestion
      if [[ -f "${outputdir}/${i}/${certificate_file}" && -f "${outputdir}/${i}/${privatekey_file}" ]]; then
```

As of #78, you have to use `${privatekey_file}` and `${certificate_file}` respectively for `cert.pem` and `key.pem`.

File: bin/dump.sh [hunk 2]
```diff
@@ -157,6 +160,28 @@ combine_pkcs12() {
...
+        log "Converting key for domain ${i} to RSA key file"
+        openssl rsa -in ${outputdir}/"${i}"/key.pem -out ${outputdir}/"${i}"/rsakey.pem
+      fi
+    done
+  else
+    if [[ -f ${outputdir}/cert.pem && -f ${outputdir}/key.pem ]]; then
```
Comment (kereis) [inline bin/dump.sh:178]:
```suggestion
    if [[ -f "${outputdir}/${certificate_file}" && -f "${outputdir}/${privatekey_file}" ]]; then
```

See previous review comment regarding #78.

File: bin/dump.sh [hunk 3]
```diff
@@ -157,6 +160,28 @@ combine_pkcs12() {
...
+    return
+  fi
+
+  if [[ -z "${DOMAIN}" || "${#DOMAINS[@]}" -gt 1 ]]; then
+    local outputdir_subdirs=(${outputdir}/*/)
+    for subdir in "${outputdir_subdirs[@]}"; do
+      local i=$(basename "${subdir}" /)
+      if [[ -f ${outputdir}/${i}/cert.pem && -f ${outputdir}/${i}/key.pem ]]; then
+        log "Converting key for domain ${i} to RSA key file"
+        openssl rsa -in ${outputdir}/"${i}"/key.pem -out ${outputdir}/"${i}"/rsakey.pem
```
Comment (kereis) [inline bin/dump.sh:174]:
```suggestion
        openssl rsa -in "${outputdir}/${i}/${privatekey_file}" -out "${outputdir}/${i}/rsakey.pem"
```

See also previous comment. For consistency, maybe we should also introduce an environment variable for settings file name + extension of `rsakey.pem`? See https://github.com/kereis/traefik-certs-dumper/commit/cea5907733267e935a8d0b10145d672589fc30d1#diff-ed933e4c63bca6ca162f9af2745864cfea768a052ed1dd83c00f04b3b4b878e3R8 for reference.

**Review by kereis — CHANGES_REQUESTED** [review]
Thank you for applying the suggestions. 👍 

Another change that I have to request: #78 was merged recently which introduces a new way of specifying file names and extensions for both certificate and private key file. So you may adjust `convert_keys_to_rsa()` accordingly. See my review comments.

File: bin/dump.sh (hunk 3 above)
Comment (ggruening) [reply bin/dump.sh]: I am happy to take care of your suggestions concerning a new environment variable for `rsakey.pem`. I'll hope I will find the time during the next days.

**Comment by ggruening** [comment]
Hey ho, I did it similar to https://github.com/kereis/traefik-certs-dumper/pull/78, should be consistent again.

**Review by kereis — CHANGES_REQUESTED** [review]
Thanks.

One tiny thing though: Can you add your new environment variables to the table of README.md? Afterwards, we can go ahead and merge your changes.

**Comment by ggruening** [comment]
I just added the new vars to the table of README.md. Furthermore I added an usage example. Hope this helps.

**Review by kereis — APPROVED** [review]
LGTM.

I will merge this one. Thank you for your contribution. 😄 👍

**Comment by ggruening** [comment]
You're welcome. Thanks again for your work. :)

**Pull request merged**